src/ConcordiumNetSdk/Types/OnChainData.cs

```diff
@@ -0,0 +1,141 @@
+using System.Formats.Cbor;
+
+using ConcordiumNetSdk.Helpers;
+
+namespace ConcordiumNetSdk.Types;
+
+/// <summary>
+/// Represents data to be stored or that was stored on-chain as part of a transaction.
+///
+/// This can be any data which is at most <see cref="MaxLength"/>
+/// bytes, but the convention is to store a single string encoded
+/// as CBOR.
```

Comment:
I'm not sure what you mean by a "single string". The convention is to encode the data using CBOR. It can be an integer or an object as well.